Makefile

```diff
@@ -31,11 +37,13 @@ SRCFILES := jit_compile.cpp par_loop_generator.cpp
 OBJFILES := $(patsubst %,$(BLDDIR)/%,$(notdir $(SRCFILES:.cpp=-cpp.o)))
 vpath %.cpp $(SRCDIRS) 
 
-$(info "BLDDIR = $(BLDDIR)")
-$(info "OBJFILES = $(OBJFILES)")
+#$(info "BLDDIR = $(BLDDIR)")
+#$(info "OBJFILES = $(OBJFILES)")
 
 .PHONY: all
-all: builddir libdir slib dlib CLEANOBJ
+# To avoid dealing with dependency files or having stale object files, object files are always cleaned up after each build
+all: builddir libdir slib dlib
+	rm -f $(BLDDIR)/*.o
```

Comment:
If CLEANOBJ was used as one of the targets in place of using rm at the end of the build, rm can happen before libs are built and this would be a problem (encountered in practical experiments).